carbon-design-system/carbon-for-ibm-dotcom
# feat(services): adjust the translation API to accept full endpoint URL protocol

### Related Ticket(s)

DDS Issue: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7519
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-2290

### Description

- This PR modifies the URL construction within the translation API services to accept HTTPS call defined in the `data-endpoint` prop/attribute.
- This protocol checking logic allows us to define the destination of the endpoint that outputs the same JSON data structure as what's being defined in the `ibm-dotcom-translations` repo, but the data will be dynamically pulled out of Drupal instead of the manual editing process.


### Changelog

**Changed**

-  Adjust the `url` variable within the `fetchTranslation` method.

### Testing
- Pull down the changes of this PR in a test preview/application (see - [link](https://github.com/carbon-design-system/carbon-for-ibm-dotcom-web-components-test/blob/master/src/layouts/cloud-base.hbs#L216)) and define the following markup.
- Ensure that the masthead is rendering the data is from the defined endpoint.
```
<dds-masthead-container
  platform="Security"
  data-endpoint="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/translations/security-masthead"
></dds-masthead-container>
```
